commit 394978867edf3746c285491db6be41abfe325d87
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Wed Aug 14 08:38:52 2013 +0100

    sna: Check that we successfully install the required hosting midlayer
    
    Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>

diff --git a/src/sna/sna_driver.c b/src/sna/sna_driver.c
index c27deb6..83a5bba 100644
--- a/src/sna/sna_driver.c
+++ b/src/sna/sna_driver.c
@@ -430,6 +430,13 @@ static Bool sna_pre_init(ScrnInfoPtr scrn, int flags)
 		goto cleanup;
 	}
 
+	/* Sanity check */
+	if (hosted() && (sna->flags & SNA_IS_HOSTED) == 0) {
+		xf86DrvMsg(scrn->scrnIndex, X_ERROR,
+			   "Failed to setup hosted device.\n");
+		goto cleanup;
+	}
+
 	preferred_depth = sna->info->gen < 030 ? 15 : 24;
 	if (!fb_supports_depth(fd, preferred_depth))
 		preferred_depth = 24;
